### PR TITLE
improve DB indexing for squid

### PR DESCRIPTION
**Pull request by DaMandal0rian**
### **User description**
Changes to Graphql schema:
- Indexed key fields for efficient lookups
- Cardinality hints for relationship optimization
- Indexes on timestamp fields for time-based queries
- Indexes on account and operator fields for common blockchain explorer queries

___

### **PR Type**
Enhancement, Documentation

___

### **Description**
- Added indexes to key fields such as `id`, `hash`, `timestamp`, `account`, and `operator` to improve database lookup efficiency.
- Consolidated `@derivedFrom` and `@cardinality` annotations on single lines for better readability.
- Enhanced indexing on timestamp fields to optimize time-based queries.
- Improved indexing on account and operator fields to support common blockchain explorer queries.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>schema.graphql</strong><dd><code>Improve database indexing and optimize GraphQL schema</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

indexers/squid-blockexplorer/schema.graphql

<li>Added indexes to multiple fields for efficient lookups.<br> <li> Consolidated <code>@derivedFrom</code> and <code>@cardinality</code> annotations on single <br>lines.<br> <li> Added <code>@index</code> to timestamp fields for time-based queries.<br> <li> Added <code>@index</code> to account and operator fields for common blockchain <br>explorer queries.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/751/files#diff-93b506a298c74c0e346ef0a991abd97f4a7a319d5804dd5f92dd3b897d84dc63">+19/-28</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: indexers/squid-blockexplorer/schema.graphql [hunk 1]
```diff
@@ -45,11 +37,11 @@ type ExtrinsicModuleName @entity {
 
 type Extrinsic @entity {
   id: ID! @index
-  hash: String!
+  hash: String! @index
   indexInBlock: Int!
   nonce: BigInt
   name: String!
```
Comment (marc-aurele-besner) [inline indexers/squid-blockexplorer/schema.graphql:43]: can we add a index on name please

File: indexers/squid-blockexplorer/schema.graphql [hunk 2]
```diff
@@ -70,7 +62,7 @@ type EventModuleName @entity {
 
 type Event @entity {
   id: ID! @index
-  indexInBlock: Int! @index
+  indexInBlock: Int!
   name: String!
```
Comment (marc-aurele-besner) [inline indexers/squid-blockexplorer/schema.graphql:66]: can we add a index on name

**Review by marc-aurele-besner — APPROVED** [review]
LGTM

File: indexers/squid-blockexplorer/schema.graphql (hunk 2 above)
Comment (DaMandal0rian) [reply indexers/squid-blockexplorer/schema.graphql]: @marc-aurele-besner done 🎉

**Review by marc-aurele-besner — APPROVED** [review]
Thanks 👍

**Review by jfrank-summit — APPROVED** [review]
When do we plan to re-index? Any idea if this have a significant impact on indexing speed?

**Comment by DaMandal0rian** [comment]
> When do we plan to re-index? Any idea if this have a significant impact on indexing speed?
@jfrank-summit  This will be deployed on a new dedicated DB, we have blue/green deployment setup, so will test this in the green environment. The modifications will only help with reads on astral, not the block processing/indexing.

**Comment by jfrank-summit** [comment]
> > When do we plan to re-index? Any idea if this have a significant impact on indexing speed?
> > @jfrank-summit  This will be deployed on a new dedicated DB, we have blue/green deployment setup, so will test this in the green environment. The modifications will only help with reads on astral, not the block processing/indexing.

Regarding index speed I meant negative impact.

**Pull request merged**